openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py

```diff
@@ -0,0 +1,763 @@
+        deprecated_accounts = env["account.account"].search(
+            [("deprecated", "=", True), ("company_id", "=", payment.company_id.id)]
+        )
+        deprecated_accounts.deprecated = False
+        try:
+            payment._synchronize_to_moves(
+                [
+                    "date",
+                    "amount",
+                    "payment_type",
+                    "partner_type",
+                    "payment_reference",
+                    "is_internal_transfer",
+                    "currency_id",
+                    "partner_id",
+                    "destination_account_id",
+                    "partner_bank_id",
+                    "journal_id",
+                ]
+            )
+        except Exception as e:
+            _logger.error("Failed for payment with id %s: %s", payment.id, e)
+            raise
+        deprecated_accounts.deprecated = True
+        to_write = {
+            "line_ids": [
+                (0, 0, line_vals)
+                for line_vals in payment._prepare_move_line_default_vals(
+                    write_off_line_vals=False
+                )
+            ]
+        }
+        payment.move_id.with_context(skip_account_move_synchronization=True).write(
+            to_write
+        )
+    # restore lock dates
+    for rc_id in company_dates:
+        env.cr.execute(
+            """
+            UPDATE res_company
+            SET period_lock_date = {}, fiscalyear_lock_date = {}
+            WHERE id = {}""".format(
+                "TO_DATE("
+                + company_dates[rc_id][0].strftime("%y-%m-%d")
+                + ", 'YYYY-MM-DD')"
+                if company_dates[rc_id][0]
+                else "NULL",
+                "TO_DATE("
+                + company_dates[rc_id][1].strftime("%y-%m-%d")
+                + ", 'YYYY-MM-DD')"
+                if company_dates[rc_id][1]
+                else "NULL",
+                rc_id,
+            ),
+        )
+    env["res.company"].invalidate_cache(
+        ["period_lock_date", "fiscalyear_lock_date"], list(company_dates.keys())
+    )
+
+
+def try_delete_noupdate_records(env):
+    openupgrade.delete_records_safely_by_xml_id(
+        env,
+        [
+            "account.sequence_payment_customer_invoice",
+            "account.sequence_payment_customer_refund",
+            "account.sequence_payment_supplier_invoice",
+            "account.sequence_payment_supplier_refund",
+            "account.sequence_payment_transfer",
+        ],
+    )
+
+
+def fill_account_move_line_amounts(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move_line aml
+        SET amount_currency = aml.debit-aml.credit
+        FROM account_move am
+        WHERE aml.currency_id IS NOT NULL AND
+            aml.move_id = am.id AND am.move_type != 'entry' AND
+            aml.debit + aml.credit > 0 AND (
+                aml.amount_currency = 0 OR aml.amount_currency IS NULL)""",
```

Comment:
After testing, this did not work well as entries in Foreign currencies with amount_currency=0 where due to difference of exchange rate and therefore should stay at 0.

In my case, it seems that I only had to fix the account_move_lines which were in the company currency.

I will test with this:

```sql
UPDATE account_move_line aml
SET amount_currency = aml.debit-aml.credit
FROM res_company rco
WHERE
	aml.currency_id is not null AND
	(aml.amount_currency=0 or aml.amount_currency is NUll) AND
	aml.balance != 0  AND
	aml.currency_id = rco.currency_id
```

I will let you know about the results of my test later, but I wanted to let you know now that the current solution is not a good one.

Reply:
I made a second version of this query, could you check?

Reply:
I think that your new query works, but you don't need anymore the table `account_move am`